packages/sensors_plus/example/lib/main.dart

```diff
@@ -170,6 +218,18 @@ class _MyHomePageState extends State<MyHomePage> {
             _magnetometerValues = <double>[event.x, event.y, event.z];
           });
         },
+        onError: (e) {
+          showDialog(
+              context: context,
+              builder: (context) {
+                return const AlertDialog(
+                  title: Text("Sensor Not Found"),
+                  content: Text(
+                      "It seems that your device doesn't support Magnetometer Sensor"),
+                );
+              });
+        },
+        cancelOnError: true,
```

Comment:
We are using [method_channel_sensor](https://github.com/fluttercommunity/plus_plugins/blob/main/packages/sensors_plus/sensors_plus_platform_interface/lib/src/method_channel_sensors.dart) of sensor_plus_platform_interface.
This old code doesn't throw for unsupported.
Therefore, in order to respond to the new example code, a class that inherits from [SensorsPlatform](https://github.com/fluttercommunity/plus_plugins/blob/main/packages/sensors_plus/sensors_plus_platform_interface/lib/sensors_plus_platform_interface.dart) must be implemented.